File: Aero Linea/Aerolinea.API/Aerolinea.API/DTOs/HandshakeRequestDTO.cs
using System.Text.Json.Serialization;

namespace Aerolinea.API.DTOs
{
    /// <summary>
    /// DTO de peticion para iniciar el proceso de autenticacion por handshake con una agencia externa.
    /// Contiene el token de entrada proporcionado por la agencia y su URL de origen.
    /// </summary>
    public class HandshakeRequestDTO
    {
        [JsonPropertyName("token_entrada")]
        public string TokenEntrada { get; set; } = string.Empty;

        [JsonPropertyName("url_agencia")]
        public string UrlAgencia { get; set; } = string.Empty;
    }

    /// <summary>
    /// DTO de respuesta devuelto al completar el handshake con una agencia externa.
    /// Contiene el token de salida que la agencia debe utilizar en las siguientes peticiones,
    /// y el porcentaje de ganancia que la agencia tiene configurado.
    /// </summary>
    public class HandshakeResponseDTO
    {
        [JsonPropertyName("token_salida")]
        public string TokenSalida { get; set; } = string.Empty;

        [JsonPropertyName("porcentajeGanancia")]
        public decimal PorcentajeGanancia { get; set; } = 0;
    }
}

File: Cadena de Hoteles/Hotel/src/main/java/org/example/dtos/HandshakeRequestDTO.java
package org.example.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * DTO con los datos enviados por una agencia externa para iniciar el proceso de handshake.
 * Los campos se mapean desde snake_case del JSON mediante anotaciones de Jackson.
 */
public class HandshakeRequestDTO {

    @JsonProperty("token_entrada")
    private String tokenEntrada;

    @JsonProperty("url_agencia")
    private String urlAgencia;

    /**
     * Retorna el token de entrada proporcionado por la agencia para autenticarse.
     * @return token de entrada.
     */
    public String getTokenEntrada() { return tokenEntrada; }

    /**
     * Asigna el token de entrada proporcionado por la agencia para autenticarse.
     * @param tokenEntrada token de entrada.
     */
    public void setTokenEntrada(String tokenEntrada) { this.tokenEntrada = tokenEntrada; }

    /**
     * Retorna la URL base del sistema de la agencia externa.
     * @return URL de la agencia.
     */
    public String getUrlAgencia() { return urlAgencia; }

    /**
     * Asigna la URL base del sistema de la agencia externa.
     * @param urlAgencia URL de la agencia.
     */
    public void setUrlAgencia(String urlAgencia) { this.urlAgencia = urlAgencia; }
}